src/Adapter/ContainerBuilder.php

```diff
@@ -134,14 +134,14 @@ public function buildContainer($containerName)
         $container = $this->loadDumpedContainer();
         if (null === $container) {
             $container = $this->compileContainer();
+        } else {
+            $this->loadModulesAutoloader($container);
```

Comment:
That's the only part I'm not sure to understand, why did you need to load this before we set `shop` end `employee` definitions?

Reply:
The `shop` and `employee` services being set in the container or not should make no difference to the autoloaders. Previously, the `loadModulesAutoloader` method was called after injecting these instances into the container regardless of whether the container was loaded from the cache or recompiled. Now, it is only needed if the cache was fresh, because otherwise the autoloaders are included in the `compileContainer` method (and I wanted to avoid unnecessarily calling `loadModulesAutoloader` twice). We could introduce an additional variable storing the result of the `null === $container` conditional and add another `if` statement after injecting the synthetic services, but I do not think it is necessary.